benchmarks/instruction_counts/execution/runner.py

```diff
@@ -0,0 +1,240 @@
+import math
+import multiprocessing
+import subprocess
+import textwrap
+import threading
+import time
+from typing import Dict, List, Optional, Set, Tuple, Union
+
+from execution.work import PYTHON_CMD, SHELL, InProgress, WorkOrder
+from worker.main import WorkerFailure, WorkerOutput
+
+
+CPU_COUNT: int = multiprocessing.cpu_count()
+
+
+class WorkerFailed(Exception):
+    """Raised in the main process when a worker failure is detected."""
+    def __init__(self, wrapped_trace: Optional[str] = None) -> None:
+        self.wrapped_trace: Optional[str] = wrapped_trace
+        super().__init__()
+
+
+class CorePool:
```

Comment:
So my understanding of this class is that it hands out CPUs (by ID) in blocks which can be returned. It seems like you go out of your way to get contiguous blocks of IDs. I don't see anything specifically wrong with that, though I also don't know if contiguous numbers are particularly important either. What's the thought process here?

Reply:
The TL;DR is it's to reduce the chance of spanning two different CPU caches / NUMA nodes unnecessarily. (Added a docstring to that effect as well.)